### PR TITLE
feat(ecs): service discovery configuration

### PR DESCRIPTION
**Pull request by clareliguori**
Related to https://github.com/spinnaker/clouddriver/pull/3604

Screenshot

### CODE REVIEW

**Comment by clareliguori** [comment]
Dependent changes are now merged

cc @anotherchrisberry

File: app/scripts/modules/ecs/src/serviceDiscovery/serviceDiscovery.read.service.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  public listServiceDiscoveryRegistries(): ng.IPromise<IServiceDiscoveryRegistryDescriptor[]> {
+    return API.all('ecs')
+      .all('serviceDiscoveryRegistries')
+      .getList();
+  }
+}
+
+export const ECS_SERVICE_DISCOVERY_READ_SERVICE = 'spinnaker.ecs.servicediscovery.read.service';
+
+module(ECS_SERVICE_DISCOVERY_READ_SERVICE, []).service('serviceDiscoveryReader', ServiceDiscoveryReader);
```
Comment (anotherchrisberry) [inline app/scripts/modules/ecs/src/serviceDiscovery/serviceDiscovery.read.service.ts]: minor thing - you don't really need this to be an Angular service...if you make the one method static, you can slim the whole thing down to:
```typescript
import { IPromise } from 'angular';
import { API } from '@spinnaker/core';
import { IServiceDiscoveryRegistryDescriptor } from './IServiceDiscovery';

export class ServiceDiscoveryReader {
  public static listServiceDiscoveryRegistries(): IPromise<IServiceDiscoveryRegistryDescriptor[]> {
    return API.all('ecs')
      .all('serviceDiscoveryRegistries')
      .getList();
  }
}
```
then, where you need it, just import the service and call the method a la `ServiceDiscoveryReader.listServiceDiscoveryRegistries()`.

**Comment by anotherchrisberry** [comment]
It would be nice to write most of this as React components, but that's asking a lot, so I think this is good!

File: app/scripts/modules/ecs/src/serviceDiscovery/serviceDiscovery.read.service.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import { API } from '@spinnaker/core';
+import { IServiceDiscoveryRegistryDescriptor } from './IServiceDiscovery';
+
+export class ServiceDiscoveryReader {
+  public static listServiceDiscoveryRegistries(): IPromise<IServiceDiscoveryRegistryDescriptor[]> {
+    return API.all('ecs')
+      .all('serviceDiscoveryRegistries')
+      .getList();
+  }
+}
```
Comment (anotherchrisberry) [inline app/scripts/modules/ecs/src/serviceDiscovery/serviceDiscovery.read.service.ts]:
```suggestion
}

```

Comment (anotherchrisberry) [reply app/scripts/modules/ecs/src/serviceDiscovery/serviceDiscovery.read.service.ts]: (super minor, add a newline at the end of the file)

**Comment by anotherchrisberry** [comment]
one last super minor thing, then we'll get this merged

**Pull request merged**